### PR TITLE
Add Azure Analysis & Databases

### LINKED ISSUE #333: [Azure] Support analysis & databases
Azure should support more resources
*   `analysis`
    * `azurerm_analysis_services_server`
*   `database`
	* `azurerm_mariadb_configuration`
	* `azurerm_mariadb_database`
	* `azurerm_mariadb_firewall_rule`
	* `azurerm_mariadb_server`
	* `azurerm_mariadb_virtual_network_rule`
	* `azurerm_mysql_configuration`
	* `azurerm_mysql_database`
	* `azurerm_mysql_firewall_rule`
	* `azurerm_mysql_server`
	* `azurerm_mysql_virtual_network_rule`
	* `azurerm_postgresql_configuration`
	* `azurerm_postgresql_database`
	* `azurerm_postgresql_firewall_rule`
	* `azurerm_postgresql_server`
	* `azurerm_postgresql_virtual_network_rule`
	* `azurerm_sql_database`
	* `azurerm_sql_active_directory_administrator`
	* `azurerm_sql_elasticpool`
	* `azurerm_sql_failover_group`
	* `azurerm_sql_firewall_rule`
	* `azurerm_sql_server`
	* `azurerm_sql_virtual_network_rule`

### PR DESCRIPTION
**Pull request by cucxabong**
This PR add more resources into Azure provider
Fix #333 
*   `analysis`
    * `azurerm_analysis_services_server`
*   `database`
	* `azurerm_mariadb_configuration`
	* `azurerm_mariadb_database`
	* `azurerm_mariadb_firewall_rule`
	* `azurerm_mariadb_server`
	* `azurerm_mariadb_virtual_network_rule`
	* `azurerm_mysql_configuration`
	* `azurerm_mysql_database`
	* `azurerm_mysql_firewall_rule`
	* `azurerm_mysql_server`
	* `azurerm_mysql_virtual_network_rule`
	* `azurerm_postgresql_configuration`
	* `azurerm_postgresql_database`
	* `azurerm_postgresql_firewall_rule`
	* `azurerm_postgresql_server`
	* `azurerm_postgresql_virtual_network_rule`
	* `azurerm_sql_database`
	* `azurerm_sql_active_directory_administrator`
	* `azurerm_sql_elasticpool`
	* `azurerm_sql_failover_group`
	* `azurerm_sql_firewall_rule`
	* `azurerm_sql_server`
	* `azurerm_sql_virtual_network_rule`

### CODE REVIEW

File: providers/azure/analysis.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/hashicorp/go-azure-helpers/authentication"
+	"log"
+)
+
+type AnalysisGenerator struct {
+	AzureService
+}
+
+func (g *AnalysisGenerator) listServiceServers() ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting Service Servers")
```
Comment (sergeylanzman) [inline providers/azure/analysis.go]: please delete log.Println here

File: providers/azure/database.go [hunk 2]
```diff
@@ -0,0 +1,895 @@
...
+	Servers, err := Client.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return *Servers.Value, nil
+}
+
+func (g *DatabasesGenerator) createMariaDBServerResources(Servers []mariadb.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MariaDB Servers")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 3]
```diff
@@ -0,0 +1,895 @@
...
+			map[string]interface{}{
+				"administrator_login_password": "",
+			}))
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMariaDBConfigurationResources(Servers []mariadb.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MariaDB Configurations")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 4]
```diff
@@ -0,0 +1,895 @@
...
+				g.ProviderName,
+				[]string{"value"}))
+		}
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMariaDBDatabaseResources(Servers []mariadb.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MariaDB Database")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 5]
```diff
@@ -0,0 +1,895 @@
...
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMariaDBFirewallRuleResources(Servers []mariadb.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MariaDB Firewall Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 6]
```diff
@@ -0,0 +1,895 @@
...
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMariaDBVirtualNetworkRuleResources(Servers []mariadb.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MariaDB Virtual Network Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 7]
```diff
@@ -0,0 +1,895 @@
...
+	Servers, err := Client.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return *Servers.Value, nil
+}
+
+func (g *DatabasesGenerator) createMySQLServerResources(Servers []mysql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MySQL Servers")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 8]
```diff
@@ -0,0 +1,895 @@
...
+			map[string]interface{}{
+				"administrator_login_password": "",
+			}))
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMySQLConfigurationResources(Servers []mysql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MySQL Configurations")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 9]
```diff
@@ -0,0 +1,895 @@
...
+				g.ProviderName,
+				[]string{"value"}))
+		}
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMySQLDatabaseResources(Servers []mysql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MySQL Databases")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 10]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_mysql_database",
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMySQLFirewallRuleResources(Servers []mysql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MySQL Firewall Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 11]
```diff
@@ -0,0 +1,895 @@
...
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createMySQLVirtualNetworkRuleResources(Servers []mysql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting MySQL VirtualNetwork Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 12]
```diff
@@ -0,0 +1,895 @@
...
+	Servers, err := Client.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return *Servers.Value, nil
+}
+
+func (g *DatabasesGenerator) createPostgreSQLServerResources(Servers []postgresql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting Postgresql Servers")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 13]
```diff
@@ -0,0 +1,895 @@
...
+			map[string]interface{}{
+				"administrator_login_password": "",
+			}))
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createPostgreSQLDatabaseResources(Servers []postgresql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting PostgreSQL Databases")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 14]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_postgresql_database",
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createPostgreSQLConfigurationResources(Servers []postgresql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting PostgreSQL Configurations")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 15]
```diff
@@ -0,0 +1,895 @@
...
+	var resources []terraform_utils.Resource
+	ctx := context.Background()
+	SubscriptionID := g.Args["config"].(authentication.Config).SubscriptionID
+	Authorizer := g.Args["authorizer"].(autorest.Authorizer)
+	//terraform_utils.NewSim
```
Comment (sergeylanzman) [inline providers/azure/database.go]: delete

File: providers/azure/database.go [hunk 16]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_postgresql_configuration",
+				g.ProviderName,
+				[]string{"value"}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createPostgreSQLFirewallRuleResources(Servers []postgresql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting PostgreSQL Firewall Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 17]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_postgresql_firewall_rule",
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createPostgreSQLVirtualNetworkRuleResources(Servers []postgresql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting PostgreSQL VirtualNetwork Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 18]
```diff
@@ -0,0 +1,895 @@
...
+		if err := ServerPages.NextWithContext(ctx); err != nil {
+			return nil, err
+		}
+	}
+
+	return servers, nil
+}
+
+func (g *DatabasesGenerator) createSQLServerResources(Servers []sql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting SQL Servers")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 19]
```diff
@@ -0,0 +1,895 @@
...
+			map[string]interface{}{
+				"administrator_login_password": "",
+			}))
+	}
+
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createSQLDatabaseResources(Servers []sql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting SQL Databases")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 20]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_sql_database",
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createSQLFirewallRuleResources(Servers []sql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting SQL Firewall Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 21]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_sql_firewall_rule",
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createSQLVirtualNetworkRuleResources(Servers []sql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting SQL VirtualNetwork Rules")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 22]
```diff
@@ -0,0 +1,895 @@
...
+		}
+		if err := ruleIter.NextWithContext(ctx); err != nil {
+			return nil, err
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createSQLElasticPoolResources(Servers []sql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting SQL Elastic Pools")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 23]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_sql_elasticpool",
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createSQLFailoverResources(Servers []sql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting SQL Failover Groups")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

File: providers/azure/database.go [hunk 24]
```diff
@@ -0,0 +1,895 @@
...
+				"azurerm_sql_failover_group",
+				g.ProviderName,
+				[]string{}))
+		}
+	}
+	return resources, nil
+}
+
+func (g *DatabasesGenerator) createSQLADAdministratorResources(Servers []sql.Server) ([]terraform_utils.Resource, error) {
+	log.Println("\tImporting SQL Active Directory Administrator")
```
Comment (sergeylanzman) [inline providers/azure/database.go]: please delete log.Println here

**Comment by cucxabong** [comment]
Fixed @sergeylanzman . Thank you

**Pull request merged**